test/reporters/tests/json.test.ts

```diff
@@ -0,0 +1,135 @@
+import { afterEach, expect, test, vi } from 'vitest'
+import { JsonReporter } from 'vitest/src/node/reporters/json'
+import type { ModuleGraph, ModuleNode, TransformResult } from 'vite'
+import { getContext } from '../src/context'
+import { clearSourcePos, files } from '../src/data'
+
+const no_comment: TransformResult = {
+  code: 'const __vite_ssr_import_0__ = await __vite_ssr_import__("vitest");\n'
```

Comment:
This is probably too verbose. I am thinking to spawn a Vitest runner and read it's JSON output to snap